packages/editor-ui/src/store.ts

```diff
@@ -214,20 +214,23 @@ export const store = new Vuex.Store({
 
 		// Pin data
 		pinData(state, payload: { node: INodeUi, data: IPinData[string] }) {
-			if (state.workflow.pinData) {
-				Vue.set(state.workflow.pinData, payload.node.name, payload.data);
+			if (!state.workflow.pinData) {
+				Vue.set(state.workflow, 'pinData', {});
 			}
 
+			Vue.set(state.workflow.pinData!, payload.node.name, payload.data);
 			state.stateIsDirty = true;
 
 			dataPinningEventBus.$emit('pin-data', { [payload.node.name]: payload.data });
 		},
 		unpinData(state, payload: { node: INodeUi }) {
-			if (state.workflow.pinData) {
-				Vue.set(state.workflow.pinData, payload.node.name, undefined);
-				delete state.workflow.pinData[payload.node.name];
+			if (!state.workflow.pinData) {
+				Vue.set(state.workflow, 'pinData', {});
 			}
 
+			Vue.set(state.workflow.pinData!, payload.node.name, undefined);
```

Comment:
Non-actionable: Would be nice if `Vue.set()` were smart enough to create missing intermediate properties to avoid the boilerplate. Also I wish TS were able to pick up that `state.workflow.pinData` is defined via `Vue.set()`.